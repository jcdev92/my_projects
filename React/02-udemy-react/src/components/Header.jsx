import React from 'react';
import PropTypes from 'prop-types';

// PropTypes are used to validate the data type of the props passed to the component

function Header({title, subtitle}) {
    return (
        <section>
            <h1 data-testid="test-title">{title}</h1>
            <h2>{subtitle}</h2>
            <h2>{subtitle}</h2>
            <h2>{subtitle}</h2>
        </section>
    );
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

// Default props are used to set default values for the props if they are not passed to the component

Header.defaultProps = {
    title: 'Default title',
    subtitle: 0
}

export default Header;