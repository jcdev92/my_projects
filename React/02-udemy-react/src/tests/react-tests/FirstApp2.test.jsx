import {render, screen} from "@testing-library/react";
import Header from "../../components/Header.jsx";


describe('Pruebas en <Header/>', () => {
    const title = 'Hola, soy Goku';
    const subtitle = 15;

    test('debe de hacer match con el snapshot', () => {
        const {container} = render(<Header title={title} subtitle={subtitle}/>);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mesaje "Hola, soy Goku"', () => {
        render(<Header title={title} />);
        expect(screen.getByText(title)).toBeTruthy();

    })

    test('debe de mostrar el titulo en h1' , () => {
        render(<Header title={title} />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    })

    test('debe de mostrar el subtitulo enviado por props' , () => {
        render(<Header title={title} subtitle={subtitle} />);
        expect(screen.getAllByText(subtitle).length).toBe(3);
    })

})