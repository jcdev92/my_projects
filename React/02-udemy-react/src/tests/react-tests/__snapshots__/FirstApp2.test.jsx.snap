// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Header/> debe de hacer match con el snapshot 1`] = `
<div>
  <section>
    <h1
      data-testid="test-title"
    >
      Hola, soy Goku
    </h1>
    <h2>
      15
    </h2>
    <h2>
      15
    </h2>
    <h2>
      15
    </h2>
  </section>
</div>
`;
